docs/messageData.md

```diff
@@ -1025,3 +1030,496 @@ the organizer forgets to scan an attendee’s public key.
+                    "voting_method": {
+                        "type": "string",
+                        "enum": ["Plurality", "Approval"],
+                        "$comment": "supported voting method"
+                    },
+                    "ballot_options": {
+                        "description": "[Array[String]] ballot options",
+                        "type": "array",
+                        "items": {
+                            "type": "string",
+                            "$comment": "ballot option"
+                        },
+                        "minItems": 2,
+                        "uniqueItems": true
+                    },
+                    "write_in": {
+                        "type": "boolean",
+                        "$comment": "whether write-in is allowed"
+                    }
+                },
+                "additionalProperties": false,
+                "required": [
+                    "id",
+                    "question",
+                    "voting_method",
+                    "ballot_options",
+                    "write_in"
+                ]
+            },
+            "minItems": 1,
+            "uniqueItems": true
+        }
+    },
+    "additionalProperties": false,
+    "required": [
+        "object",
+        "action",
+        "id",
+        "lao",
+        "name",
+        "version",
+        "created_at",
+        "start_time",
+        "end_time",
+        "questions"
+    ]
+}
+
+```
+
+## Casting a vote (election#cast_vote)
+
+🧭 **RPC Message** > **RPC payload** (*Query*) > **Query payload** (*Publish*) >
+**Mid Level** > **High level** (*election#cast_vote*)
+
+A member of the LAO can cast a vote by publishing an election/cast_vote message to the
+election’s channel. Each member may cast multiple votes, only the last one will be counted.
+If write-in is allowed for the election then the vote has to have a write-in.
+
+<details>
+<summary>
+💡 See some examples
+</summary>
+
+```json5
+// ../protocol/examples/messageData/vote_cast_vote.json
+
+{
+    "object": "election",
+    "action": "cast_vote",
+    "lao": "fzJSZjKf-2cbXH7kds9H8NORuuFIRLkevJlN7qQemjo=",
+    "election": "zG1olgFZwA0m3mLyUqeOqrG0MbjtfqShkyZ6hlyx1tg=",
+    "created_at": 1633098941,
+    "votes": [
+        {
+            "id": "8L2MWJJYNGG57ZOKdbmhHD9AopvBaBN26y1w5jL07ms=",
+            "question": "2PLwVvqxMqW5hQJXkFpNCvBI9MZwuN8rf66V1hS-iZU=",
+            "vote": [0]
+        }
+    ]
+}
+
+```
+```json5
+// ../protocol/examples/messageData/vote_cast_write_in.json
+
+{
+    "object": "election",
+    "action": "cast_vote",
+    "lao": "fzJSZjKf-2cbXH7kds9H8NORuuFIRLkevJlN7qQemjo=",
+    "election": "QWTmcWMMMiUdWdZX7ib7GyqH6A5ifDYwPaMpKxIZm1k=",
+    "created_at": 1633098996,
+    "votes": [
+        {
+            "id": "DtIsj7nQ0Y4iLJ4ETKv2D0uah7IYGyEVW7aCLFjaL0w=",
+            "question": "WBVsWJI-C5YkD0wdE4DxnLa0lJzjnHEd67XPFVB9v3g=",
+            "write_in": "Computer Science"
```

Comment:
isn't this functionality unsupported in practice?

Reply:
Yes indeed. Should I explain that write-in is unsupported in practice or delete the example?